pkg/smokescreen/config_loader.go

```diff
@@ -151,6 +156,23 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	c.TimeConnect = yc.TimeConnect
 	c.UnsafeAllowPrivateRanges = yc.UnsafeAllowPrivateRanges
 
+	if yc.MitmCaCertFile != "" || yc.MitmCaKeyFile != "" {
+		if yc.MitmCaCertFile == "" {
+			return errors.New("mitm_ca_cert_file required when mitm_ca_key_file is set")
+		}
+		if yc.MitmCaKeyFile == "" {
+			return errors.New("mitm_ca_key_file required when mitm_ca_cert_file is set")
+		}
+		mitmCa, err := tls.LoadX509KeyPair(yc.MitmCaCertFile, yc.MitmCaKeyFile)
+		if err != nil {
+			return fmt.Errorf("could not load mitmCa: %v", err)
+		}
+		if mitmCa.Leaf, err = x509.ParseCertificate(mitmCa.Certificate[0]); err != nil {
```

Comment:
Do we need a `len()` check here before we index into it? What if it is >1?

Reply:
Yes definitively.
`mitmCa.Leaf` should only contain the leaf cert and no intermediate certs